misc/multistream-select/src/dialer_select.rs

```diff
@@ -44,15 +43,13 @@ use std::{
 /// all versions supported by this library. Frictionless multistream-select
 /// protocol upgrades may thus proceed by deployments with updated listeners,
 /// eventually followed by deployments of dialers choosing the newer protocol.
-pub fn dialer_select_proto<R, I>(
+pub fn dialer_select_proto<R>(
     inner: R,
-    protocols: I,
+    protocols: Vec<Protocol>,
```

Comment:
What is the rational for requiring a `Vec` instead of an `IntoInterator` here and then calling `into_iter` on it right away in line prev. 57 now 54?

Reply:
Originally I thought it simplifies the API but the allocation is actually unnecessary so I agree that this should be reverted.

Reply:
:+1: for reverting.